pkg/apis/componentconfig/types.go

```diff
@@ -62,3 +62,213 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+type KubeletConfiguration struct {
```

Comment:
IIUC omitempty has no effect on the deserialize path. We won't be deserialize -> serializing or desiralize -> serialize -> desirializing this object for now at least. We will only be deserializing it. This object also won't ever be an apiserver resource so swagger is not a useful documentation mechanism for it. So maybe it doesn't matter yet?